lib/Sema/TypeCheckStmt.cpp

```diff
@@ -2366,13 +2366,15 @@ bool TypeCheckASTNodeAtLocRequest::evaluate(
         }
       }
     } else if (auto *defaultArg = dyn_cast<DefaultArgumentInitializer>(DC)) {
+      ParamDecl *Param = nullptr;
       if (auto *AFD = dyn_cast<AbstractFunctionDecl>(defaultArg->getParent())) {
-        auto *Param = AFD->getParameters()->get(defaultArg->getIndex());
-        (void)Param->getTypeCheckedDefaultExpr();
-        return false;
+        Param = AFD->getParameters()->get(defaultArg->getIndex());
+      } else if (auto *MD = dyn_cast<MacroDecl>(defaultArg->getParent())) {
+        Param = MD->getParameterList()->get(defaultArg->getIndex());
+      } else if (auto *SD = dyn_cast<SubscriptDecl>(defaultArg->getParent())) {
+        Param = SD->getIndices()->get(defaultArg->getIndex());
```

Comment:
Can this use `getParameterAt` instead? `swift::getParameterList` already handles `MacroDecl` too.